File: 003/Haskell/Problem003Test.hs
module Main where

import Test.HUnit
import System.Exit
import Problem003

greatest_prime_factor_tests = TestList [
  TestCase (assertEqual "GPF of 2" 2 (greatestPrimeFactor 2)),
  TestCase (assertEqual "GPF of 4" 2 (greatestPrimeFactor 4)),
  TestCase (assertEqual "GPF of 6" 3 (greatestPrimeFactor 6)),
  TestCase (assertEqual "GPF of 13195" 29 (greatestPrimeFactor 13195))
  ]

checkTests counts = case counts of
	Counts _ _ errors failures ->
		if (errors + failures > 0)
		then exitFailure
		else (exitWith ExitSuccess)

tests = TestList [
  TestLabel "testing GPF" greatest_prime_factor_tests
  ]

main = runTestTT tests  >>= checkTests

File: 003/Ruby/problem003_test.rb
require "test/unit"

require_relative "problem003"

class TestProblem003 < Test::Unit::TestCase
  def test_greatest_prime_factor
    assert_equal 2, 2.greatest_prime_factor
    assert_equal 2, 4.greatest_prime_factor
    assert_equal 3, 6.greatest_prime_factor
    assert_equal 29, 13195.greatest_prime_factor
  end
end
